src/main/java/bio/terra/workspace/service/iam/ControlledResourceInheritanceMapping.java

```diff
@@ -0,0 +1,87 @@
+ * https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf
+ *
+ * <p>This mapping is slightly different for each category of controlled resource (user-shared,
+ * user-private, app-shared, app-private), so this class holds four separate maps.
+ */
+public class ControlledResourceInheritanceMapping {
+  // Currently, shared resources have the same permission inheritance regardless of whether they
+  // are user- or application-controlled. This list is pulled out separately as a convenience.
+  private static ImmutableMap<WsmIamRole, ImmutableList<ControlledResourceIamRole>>
+      SHARED_RESOURCE_MAPPING =
+          ImmutableMap.of(
+              WsmIamRole.OWNER,
+              ImmutableList.of(ControlledResourceIamRole.EDITOR, ControlledResourceIamRole.WRITER),
+              WsmIamRole.WRITER,
+              ImmutableList.of(ControlledResourceIamRole.EDITOR, ControlledResourceIamRole.WRITER),
+              // Applications are granted permissions individually, not as a role.
+              WsmIamRole.APPLICATION,
+              ImmutableList.of(),
+              WsmIamRole.READER,
+              ImmutableList.of(ControlledResourceIamRole.READER));
+
+  public static ImmutableMap<WsmIamRole, ImmutableList<ControlledResourceIamRole>>
+      USER_SHARED_MAPPING = SHARED_RESOURCE_MAPPING;
+
+  public static ImmutableMap<WsmIamRole, ImmutableList<ControlledResourceIamRole>>
+      APPLICATION_SHARED_MAPPING = SHARED_RESOURCE_MAPPING;
+
+  public static ImmutableMap<WsmIamRole, ImmutableList<ControlledResourceIamRole>>
+      USER_PRIVATE_MAPPING =
+          ImmutableMap.of(
+              WsmIamRole.OWNER,
+              ImmutableList.of(
+                  ControlledResourceIamRole.ASSIGNER, ControlledResourceIamRole.EDITOR),
+              WsmIamRole.WRITER,
+              ImmutableList.of(ControlledResourceIamRole.EDITOR),
+              // Applications and readers have no permissions on private resources.
+              WsmIamRole.APPLICATION,
+              ImmutableList.of(),
+              WsmIamRole.READER,
+              ImmutableList.of());
+
+  public static ImmutableMap<WsmIamRole, ImmutableList<ControlledResourceIamRole>>
+      APPLICATION_PRIVATE_MAPPING =
+          ImmutableMap.of(
+              WsmIamRole.OWNER, ImmutableList.of(ControlledResourceIamRole.EDITOR),
+              WsmIamRole.WRITER, ImmutableList.of(ControlledResourceIamRole.EDITOR),
+              // Applications and readers have no permissions on private resources.
+              WsmIamRole.APPLICATION, ImmutableList.of(),
+              WsmIamRole.READER, ImmutableList.of());
+
+  public static ImmutableMap<WsmIamRole, ImmutableList<ControlledResourceIamRole>>
+      getInheritanceMapping(AccessScopeType accessScope, ManagedByType managedBy) {
+    if (accessScope == AccessScopeType.ACCESS_SCOPE_SHARED) {
+      if (managedBy == ManagedByType.MANAGED_BY_USER) {
+        return USER_SHARED_MAPPING;
```

Comment:
do these still need to be public if this is the right way to access them? Should they just be in a Table<AccessScopeType, ManagedByType, Map> instead?

Reply:
done. building a table for 4 elements seems a little overkill, but it could reasonably go either way